ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -197,17 +199,35 @@ public TransactionSelectionResults buildTransactionListForBlock() {
   }
 
   private void timeLimitedSelection() {
+    final long startTime = System.currentTimeMillis();
+
+    selectorsStateManager.blockSelectionStarted();
+
+    pluginTimeLimitedSelection(startTime);
```

Comment:
What is the rationale of starting with plugin time limited selection before internal limited selection. From the implementation below, `pluginSelectionDone.await()` can block internal selection to happen if takes too much time to finish the ongoing transaction selection that was interrupted.
We can also imagine something like this 
```
boolean completed = pluginSelectionDone.await(PLUGIN_CLEANUP_TIMEOUT, TimeUnit.MILLISECONDS);
if (!completed) {
    LOG.warn("Plugin selection cleanup did not complete within the timeout.");
}
```